llama_index/indices/response/tree_summarize.py

```diff
@@ -1,146 +1,148 @@
-from typing import Any, Dict, List, Optional, Sequence, Tuple
+import asyncio
+from typing import Any, List, Optional, Sequence, Tuple
 
-from llama_index.data_structs.data_structs import IndexGraph
-from llama_index.data_structs.node import Node
-from llama_index.indices.common_tree.base import GPTTreeIndexBuilder
-from llama_index.indices.response.refine import Refine
+from llama_index.async_utils import run_async_tasks
+from llama_index.indices.response.base_builder import BaseResponseBuilder
 from llama_index.indices.service_context import ServiceContext
-from llama_index.indices.utils import get_sorted_node_list
+from llama_index.prompts.default_prompts import DEFAULT_TEXT_QA_PROMPT
 from llama_index.prompts.prompt_type import PromptType
-from llama_index.prompts.prompts import (
-    QuestionAnswerPrompt,
-    RefinePrompt,
-    SummaryPrompt,
-)
-from llama_index.storage.docstore.registry import get_default_docstore
+from llama_index.prompts.prompts import QuestionAnswerPrompt, SummaryPrompt
 from llama_index.token_counter.token_counter import llm_token_counter
 from llama_index.types import RESPONSE_TEXT_TYPE
 
 
-class TreeSummarize(Refine):
+class TreeSummarize(BaseResponseBuilder):
     def __init__(
         self,
-        service_context: ServiceContext,
-        text_qa_template: QuestionAnswerPrompt,
-        refine_template: RefinePrompt,
+        service_context: Optional[ServiceContext] = None,
+        text_qa_template: Optional[QuestionAnswerPrompt] = None,
         streaming: bool = False,
         use_async: bool = True,
+        verbose: bool = False,
     ) -> None:
         super().__init__(
             service_context=service_context,
-            text_qa_template=text_qa_template,
-            refine_template=refine_template,
             streaming=streaming,
         )
+        self._text_qa_template = text_qa_template or DEFAULT_TEXT_QA_PROMPT
         self._use_async = use_async
+        self._verbose = verbose
 
     @llm_token_counter("aget_response")
     async def aget_response(
         self,
         query_str: str,
         text_chunks: Sequence[str],
         prev_response: Optional[str] = None,
```

Comment:
future PR: planning on cleaning up this interface and removing this `prev_response` thing